src/daemon/command_parser_executor.cpp

```diff
@@ -334,17 +334,20 @@ bool t_command_parser_executor::set_limit(const std::vector<std::string>& args)
   if(args.size()==0) {
     return m_executor.get_limit();
   }
-  int limit;
+  int64_t limit;
   try {
-      limit = std::stoi(args[0]);
+      limit = std::stoll(args[0]);
   }
   catch(std::invalid_argument& ex) {
       return false;
   }
-  if (limit==-1)  limit=128;
+  if (limit==-1)
+  {
+    return m_executor.set_limit(nodetool::default_limit_down * 1024, nodetool::default_limit_up * 1024);
+  }
```

Comment:
Minor, but I think this'd be better if it was the same as the other two: if (limit == -1) { limit_down=xxx; limit_up = yyy; } and a single call to the executor. Though I'd move the -1 to default mapping in the server (so RPC can be told to reset defaults). There's also no "leave as is" AFAICT.

Reply:
Will move the check to core_rpc_server.cpp